sdk/src/main/java/io/dapr/actors/DaprClientBase.java

```diff
@@ -1,206 +1,128 @@
+// base class of hierarchy
+public class DaprClientBase {
+    /**
+     * Defines the standard application/json type for HTTP calls in Dapr.
+     */
+    private static final MediaType MEDIA_TYPE_APPLICATION_JSON = MediaType.get("application/json; charset=utf-8");
+
+    /**
+     * Shared object representing an empty request body in JSON.
+     */
+    private static final RequestBody REQUEST_BODY_EMPTY_JSON = RequestBody.create(MEDIA_TYPE_APPLICATION_JSON, "");
+
+    /**
+     * JSON Object Mapper.
+     */
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private final String baseUrl;
+
+    /**
+     * Http client used for all API calls.
+     */
+    private final OkHttpClient httpClient;
+
+    /**
+     * Creates a new instance of {@link DaprClientBase}.
+     * @param port Port for calling Dapr. (e.g. 3500)
+     * @param httpClient RestClient used for all API calls in this new instance.
+     */
+    public DaprClientBase(int port, OkHttpClient httpClient)
+    {
+        this.baseUrl = String.format("http://%s:%d/", Constants.DEFAULT_HOSTNAME, port);;
+        this.httpClient = httpClient;
+    }
+
+    // common methods
+    /**
+     * Invokes an API asynchronously that returns Void.
+     * @param method HTTP method.
+     * @param urlString url as String.
+     * @param json JSON payload or null.
+     * @return Asynchronous Void
+     */
+    protected final Mono<Void> invokeAPIVoid(String method, String urlString, String json) {
+        return this.invokeAPI(method, urlString, json).then();
+    }
+
+    /**
+     * Invokes an API asynchronously that returns a text payload.
+     * @param method HTTP method.
+     * @param urlString url as String.
+     * @param json JSON payload or null.
+     * @return Asynchronous text
+     */
+    protected final Mono<String> invokeAPI(String method, String urlString, String json) {
+        return Mono.fromSupplier(() -> {
+            try {
+                return tryInvokeAPI(method, urlString, json);
+            } catch (IOException e) {
+                e.printStackTrace();
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    /**
+     * Invokes an API synchronously and returns a text payload.
+     * @param method HTTP method.
+     * @param urlString url as String.
+     * @param json JSON payload or null.
+     * @return text
+     */
+    protected final String tryInvokeAPI(String method, String urlString, String json) throws IOException {
+        String requestId = UUID.randomUUID().toString();
+        RequestBody body = json != null ? RequestBody.create(MEDIA_TYPE_APPLICATION_JSON, json) : REQUEST_BODY_EMPTY_JSON;
+
+        Request request = new Request.Builder()
+                .url(new URL(this.baseUrl + urlString))
+                .method(method, body)
+                .addHeader(Constants.HEADER_DAPR_REQUEST_ID, requestId)
+                .build();
+
+        // TODO: make this call async as well.
```

Comment:
We have the code snippet for this, can you put it in?

Reply:
I think that can go separately.